### ISSUE TITLE
Cannot use MINGW32 AutotoolsProjectManager with Qt Creator 3.6.0

### ISSUE THREAD

**Issue by casper** [Alexpux/MINGW-packages#1185]
What's the correct way to use and install AutotoolsProjectManager with Qt Creator 3.6.0 (downloaded from the official Qt website)?

If I use the default 3.6.0 plugin that comes bundled with Qt Creator then the `autoreconf` step just exits with:

    20:35:43: Could not start process "autoreconf" --force --install
    Error while building/deploying project twsapi (kit: Qt 5.5.1 MSYS2 MINGW32)
    When executing step "Autoreconf"

If I try to copy the `AutotoolsProjectManager.dll` file from `MSYS2\mingw32\lib\qtcreator\plugins` into the Qt Creator plugins folder then Qt Creator fails to load it because this dll is for version 3.5.1 of Qt Creator and not 3.6.0.

How is this stuff supposed to really work? Why doesn't the bundled AutotoolsProjectManager work?

**Comment by mingwandroid** [comment]
Why not use MSYS2's qt-creator?

**Comment by casper** [comment]
It crashes with a segfault when I try to open the project (`Makefile.am`):   
https://github.com/rudimeier/twsapi

I guess I'm dreaming a bit too much trying to make this project work from within Qt Creator. I can build it OK just from the command line, but wanted the autotools/makefile integration with Qt Creator to have everything work from there. 

But I guess I have to just build it from the command line instead.

**Comment by mingwandroid** [comment]
Please file that as a bug or re-title this bug and reopen it then.

I am the defacto maintainer of the autotools plugin in qtcreator, because the author seems to have left the qt project and I am the only person to work on it significantly since then AFAIK.

**Comment by casper** [comment]
Ok..sorry. I didn't file it as a bug because I think it crashes because I ran 3.6.0 first and then ran MSYS 3.5.1. Both versions probably don't like to co-exist (they detect different kits for example).

Can I ask you why it doesn't work with the 3.6.0 version downloaded directly from the Qt website?
Everything looks OK on the project settings screen, but the plugin never asks for the paths to the autotools. It seems to assume they are already in the system path.

So basically it seems the autotools plugin can not work at all unless you run Qt Creator directly from within MINGW, is that the reason? Otherwise I think it should launch the autotools with something like `path/to/MSYS/bin/sh.exe -c "autoreconf -vfi"`. But on 3.6.0 it just tries to run them under `cmd.exe` it seems.

**Comment by mingwandroid** [comment]
You should run qt creator from an msys2 shell if you want to get anywhere with the autotools plugin.

I wonder if we should split qt creators config into a different directory.. Either that or we should get our patches all upstreamed for it. There aren't that many behaviour changing ones for qt creator though. Upstream qt creator should be able to use our kits well enough.

Their install process will overwrite the config files though so you'd need to back them up, or else just install ours afterwards to recreate them.

**Comment by casper** [comment]
Ok. It would be nice if the Win32 3.6.0 plugin would warn the user about this, or somehow inform the user that it's not going to work under cmd.exe. As a first-time Qt Creator user I spent a considerable amount of time trying to figure out what the problem was. 

The Wiki entries and sparsely strewn out docs on the web make it a bit confusing, since many assume you're running under Linux, where stuff would just work, but it's not obvious from the Wiki or doc entries which OS is discussed or targeted. 

For example, this official page (which is the first entry that comes up on Google when searching for autotools for Qt Creator) makes you think it's just a mater of enabling the plugin and everything will work out of the box. No references to any OS, MINGW, MSYS, or anything else.

http://doc.qt.io/qtcreator/creator-projects-autotools.html

Anyway I realize that page is not under MSYS2 project control, but just giving it as an example how confusing it is for a newbie. 

Of course now it's perfectly clear what I need to do, but only after you told me :)

**Comment by mingwandroid** [comment]
Ok, I suggest you file a bug over at Qt Project, it is all Open Source you know? :-)

**Comment by mingwandroid** [comment]
Anyway, I suggest using MSYS2 fully and ignoring other software.

**Comment by casper** [comment]
Yes. That seems to be the best route to take. Thanks for the help!

Will have to test 3.5.1 also to see if this was really a legit bug or just a problem with 3.6.0 + 3.5.1. I'll file a separate report if I stumble on the crash again.

**Comment by mingwandroid** [comment]
Well, it is a little buggy. That I know. Changing files behind its back while it is processing them causes it to crash for example. It may end up that the Qt Project decides to remove it altogether which would be a shame.
